packages/ui/src/Button/index.tsx

```diff
@@ -0,0 +1,177 @@
+}
+
+const StyledButton = styled.button<StyledButtonProps>`
+  ${button}
+
+  background-color: ${props => getBackgroundColor(props.$actionType, props.$priority, props.theme)};
+  border: none;
+  outline: ${props =>
+    props.$priority === 'secondary'
+      ? `1px solid ${props.theme.color[borderColorByActionType[props.$actionType]].main}`
+      : 'none'};
+  outline-offset: -1px;
+  display: flex;
+  gap: ${props => props.theme.spacing(2)};
+  align-items: center;
+  justify-content: center;
+  color: ${props => props.theme.color.neutral.contrast};
+  cursor: pointer;
+  overflow: hidden;
+  position: relative;
+
+  ${props => (props.$size === 'dense' ? dense : sparse)}
+
+  ${buttonInteractions}
+
+  &::after {
+    outline-offset: -2px;
+  }
+`;
+
+interface BaseButtonProps {
+  /**
+   * The button label. If `iconOnly` is `true`, this will be used as the
+   * `aria-label` attribute.
+   */
+  children: string;
+  /**
+   * Set to `sparse` for more loosely arranged layouts, such as when this is the
+   * submit button for a form. Use `dense` when, e.g., this button appears next
+   * to every item in a dense list of data.
+   *
+   * Default: `sparse`.
+   */
+  size?: Size;
+  /**
+   * What type of action is this button related to? Leave as `default` for most
+   * buttons, set to `accent` for the single most important action on a given
+   * page, set to `unshield` for actions that will unshield the user's funds,
+   * and set to `destructive` for destructive actions.
+   *
+   * Default: `default`
+   */
+  actionType?: ActionType;
+  disabled?: boolean;
+  onClick?: MouseEventHandler<HTMLButtonElement>;
+}
+
+interface IconOnlyProps {
+  /**
+   * When `true`, will render just an icon button. The label text passed via
+   * `children` will be used as the `aria-label`.
+   */
+  iconOnly: true;
+  /**
+   * The icon import from `lucide-react` to render. If `iconOnly` is `true`, no
+   * label will be rendered -- just the icon. Otherwise, the icon will be
+   * rendered to the left of the label.
+   *
+   * ```tsx
+   * import { ChevronRight } from 'lucide-react';
+   * <Button icon={ChevronRight}>Label</Button>
+   * <Button icon={ChevronRight} iconOnly>Label</Button>
+   * ```
+   */
+  icon: LucideIcon;
+}
+
+interface RegularProps {
+  /**
+   * When `true`, will render just an icon button. The label text passed via
+   * `children` will be used as the `aria-label`.
+   */
+  iconOnly?: false;
+  /**
+   * The icon import from `lucide-react` to render. If `iconOnly` is `true`, no
+   * label will be rendered -- just the icon. Otherwise, the icon will be
+   * rendered to the left of the label.
+   *
+   * ```tsx
+   * import { ChevronRight } from 'lucide-react';
+   * <Button icon={ChevronRight}>Label</Button>
+   * <Button icon={ChevronRight} iconOnly>Label</Button>
+   * ```
+   */
+  icon?: LucideIcon;
+}
+
+export type ButtonProps = BaseButtonProps & (IconOnlyProps | RegularProps);
```

Comment:
todo: add html-native `type` attribute and assign it to `button` as default. Otherwise it will always trigger a submit and reload if used within a form

Reply:
Done!